src/prelude/internals.mo

```diff
@@ -532,9 +532,10 @@ func @timer_helper() : async () {
       if (n.expire[0] > 0 and n.expire[0] <= now and gathered < thunks.size()) {
         thunks[gathered] := ?(n.job);
         switch (n.delay) {
-          case (?delay) if (delay != 0) {
-            // re-add the node
-            let expire = n.expire[0] + delay;
+          case (null or ?0) ();
+          case (?delay) {
+            // re-add the node, skipping past expirations
+            let expire = n.expire[0] + delay * (1 + (now - n.expire[0]) / delay);
```

Comment:
Could this maybe be simplified to `let expire = now + delay`?
I am not sure about the case of `delay == 0`: In the current logic, the expiration could then still be in the past?

Reply:
Thank you. I think I now understand. The difference arises from the integer division that brings us back the recurring grid.